framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -194,13 +194,12 @@ def evaluateGradient(self, optVarsValues, traj):
       gradArray[var] = np.zeros(2) #why are we initializing to this?
     # Evaluate gradient at each point
     # first, get average opt point
-    optOutAvg = np.average(list(optVarsValues[i*2]['output'] for i in range(self.gradDict['numIterForAve'])))
     # then, evaluate gradients
     for i in range(self.gradDict['numIterForAve']):
       opt  = optVarsValues[i*2]     #the latest opt point
       pert = optVarsValues[i*2 + 1] #the perturbed point
       #calculate grad(F) wrt each input variable
-      lossDiff = pert['output'] - optOutAvg
+      lossDiff = pert['output'] - opt['output'] #optOutAvg
```

Comment:
This appears to be reverting the change made previously.  The old method showed much poorer convergence.  Why are we reverting?